forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/bt/BTResponseParser.kt

```diff
@@ -15,48 +18,61 @@ internal class BTResponseParser(btRes: Result<Any?>) : VaultResponseParser {
 
     val isSuccessful: Boolean
 
+    private val failureResponseRegExp = BtFailureResponseRegExp(btRes)
+
     init {
         // BT returns a Result<Any?> so it's never null
         vaultError = parseVaultError(btRes)
-        forageError = parseForageError(BaseResponseRegExp(btRes))
+        forageError = parseForageError(failureResponseRegExp)
         isSuccessful = btRes.isSuccess && vaultError == null && forageError == null
         successfulResponse = parseSuccessfulResponse(btRes)
     }
 
     private fun parseVaultError(vaultResponse: BasisTheoryResponse): ForageApiResponse.Failure? {
         if (vaultResponse.isSuccess) return null
-        val resRegExp = BtResponseRegExp(vaultResponse)
-        return try {
-            // In AbstractVaultSubmitter we track the forageError
-            // and the forage status code of the UnknownErrorApiResponse
-            // via the VaultProxyResponseMonitor. This keeps us informed
-            // of *when* erroroneous BT response happen. Unfortunately,
-            // we do not currently track the specifics of the proxy_error
-            // that that BT returned to us.
-            // TODO: log the specific error that BT responds with when
-            //  resRegExp.containsProxyError == true
-            if (resRegExp.containsProxyError) UnknownErrorApiResponse else null
-        } catch (_: Exception) {
-            null
-        }
```

Comment:
The try catch block is not necessary  here because regexp matching won't throw so I remove it in this PR. To elaborate on why it won't throw:
* if the regexp don't find a match they return null and the compiler enforces that we use null-safety accessors (`?.`).
* if the regexp does match, `.get(0) always refers to the entire match and `.get(1)` refers to the first matched group 

Point being that there won't ever be an index-out-of-bounds scenario by doing `.get(1)` or any other error from not finding a regexp match. I added unit tests in `BTResponseParserText.kt` and `BTFailureResponseRegExp.kt`  to confirm

Reply:
A learning I took from implementing these changes in iOS and JS is that we should really force all error scenarios through the BT codepath to ensure we are happy with the parsing. The error scenarios are:
- EBT network error code (eg. `ebt_error_55`)
- Forage API error (eg. `invalid_merchant_header`)
- BT error (eg. IDK COULD BE ANYTHING)

Reply:
Great point! I'll incorporate this into the bug bash on Friday. Thanks for surfacing this!!